### ISSUE TITLE
Cannot install poco with --with-static

### ISSUE THREAD

**Issue by Shians** [Homebrew/homebrew-core#11133]
### Bug
```
brew install --with-static poco
==> Using the sandbox
==> Downloading https://pocoproject.org/releases/poco-1.7.8/poco-1.7.8-all.tar.gz
Already downloaded: /Users/su.s/Library/Caches/Homebrew/poco-1.7.8.tar.gz
Error: SHA256 mismatch
Expected: 0b620887bb24721c72487a17523fb09d81b459eef73ae815bd6e7d308d466dd0
Actual: 4e426b4870a79fb42511f6aeadbb688b4047bdc3c9a21ac31deb364ce49ec69f
Archive: /Users/su.s/Library/Caches/Homebrew/poco-1.7.8.tar.gz
To retry an incomplete download, remove the file above.
```
Works fine without --with-static.

## config
```
HOMEBREW_VERSION: 1.1.11
ORIGIN: https://github.com/Homebrew/brew
HEAD: 394f9fa0aaa5854aa52bc589708a079665dcf462
Last commit: 9 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 4469165031e12ee1a1e0f6155a6122a524a0e366
Core tap last commit: 5 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: octa-core 64-bit skylake
Homebrew Ruby: 2.0.0-p648
Clang: 8.0 build 800
Git: 2.10.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Perl: /usr/bin/perl
Python: /Users/su.s/anaconda3/bin/python => /Users/su.s/anaconda3/bin/python3.6
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: 1.6.0_65-b14-468
macOS: 10.12.3-x86_64
Xcode: 8.2.1
CLT: 8.2.0.0.1.1480973914
X11: 2.7.11 => /opt/X11
```

## doctor
```
Warning: Anaconda is known to frequently break Homebrew builds, including Vim and
MacVim, due to bundling many duplicates of system and Homebrew-available
tools.

If you encounter a build failure please temporarily remove Anaconda
from your $PATH and attempt the build again prior to reporting the
failure to us. Thanks!

Warning: python is symlinked to python3
This will confuse build scripts and in general lead to subtle breakage.

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and what additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew provided
script of the same name. We found the following "config" scripts:
  /Users/su.s/anaconda3/bin/curl-config
  /Users/su.s/anaconda3/bin/freetype-config
  /Users/su.s/anaconda3/bin/icu-config
  /Users/su.s/anaconda3/bin/libpng-config
  /Users/su.s/anaconda3/bin/libpng16-config
  /Users/su.s/anaconda3/bin/python3-config
  /Users/su.s/anaconda3/bin/python3.6-config
  /Users/su.s/anaconda3/bin/python3.6m-config
  /Users/su.s/anaconda3/bin/xml2-config
  /Users/su.s/anaconda3/bin/xslt-config
  /usr/local/munki/munkiwebadmin-config

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libgcc_ext.10.4.dylib
  /usr/local/lib/libgcc_ext.10.5.dylib
  /usr/local/lib/libgcc_s_x86_64.1.dylib
  /usr/local/lib/libgfortran.3.dylib
  /usr/local/lib/libgomp.1.dylib
  /usr/local/lib/libquadmath.0.dylib
  /usr/local/lib/libtcl8.6.dylib
  /usr/local/lib/libtk8.6.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/fakemysql.h
  /usr/local/include/fakepq.h
  /usr/local/include/fakesql.h
  /usr/local/include/itcl.h
  /usr/local/include/itcl2TclOO.h
  /usr/local/include/itclDecls.h
  /usr/local/include/itclInt.h
  /usr/local/include/itclIntDecls.h
  /usr/local/include/itclMigrate2TclCore.h
  /usr/local/include/itclTclIntStubsFcn.h
  /usr/local/include/mysqlStubs.h
  /usr/local/include/odbcStubs.h
  /usr/local/include/pqStubs.h
  /usr/local/include/tcl.h
  /usr/local/include/tclDecls.h
  /usr/local/include/tclOO.h
  /usr/local/include/tclOODecls.h
  /usr/local/include/tclPlatDecls.h
  /usr/local/include/tclThread.h
  /usr/local/include/tclTomMath.h
  /usr/local/include/tclTomMathDecls.h
  /usr/local/include/tdbc.h
  /usr/local/include/tdbcDecls.h
  /usr/local/include/tdbcInt.h
  /usr/local/include/tk.h
  /usr/local/include/tkDecls.h
  /usr/local/include/tkPlatDecls.h

Warning: Unbrewed .la files were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected .la files:
  /usr/local/lib/libgfortran.la
  /usr/local/lib/libgomp.la
  /usr/local/lib/libquadmath.la

Warning: Unbrewed .pc files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected .pc files:
  /usr/local/lib/pkgconfig/tcl.pc
  /usr/local/lib/pkgconfig/tk.pc

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libgfortran.a
  /usr/local/lib/libgomp.a
  /usr/local/lib/libquadmath.a
  /usr/local/lib/libtclstub8.6.a
  /usr/local/lib/libtkstub8.6.a
```

**Comment by fxcoudert** [comment]
Remove the cached download `/Users/su.s/Library/Caches/Homebrew/poco-1.7.8.tar.gz` and trying again please.

**Comment by Shians** [comment]
With cache cleared

```
brew install --with-static poco
Updating Homebrew...
==> Using the sandbox
==> Downloading https://pocoproject.org/releases/poco-1.7.8/poco-1.7.8-all.tar.g
######################################################################## 100.0%
Error: SHA256 mismatch
Expected: 0b620887bb24721c72487a17523fb09d81b459eef73ae815bd6e7d308d466dd0
Actual: 4e426b4870a79fb42511f6aeadbb688b4047bdc3c9a21ac31deb364ce49ec69f
Archive: /Users/su.s/Library/Caches/Homebrew/poco-1.7.8.tar.gz
To retry an incomplete download, remove the file above.
```

**Comment by dunn** [comment]
I'm seeing this too
```
==> Downloading https://pocoproject.org/releases/poco-1.7.8/poco-1.7.8-all.tar.gz
/usr/bin/curl --remote-time --location --user-agent Homebrew/1.1.11-98-gbfb5bf1d (Macintosh; Intel macOS 10.12.3) curl/7.51.0 --fail https://pocoproject.org/releases/poco-1.7.8/poco-1.7.8-all.tar.gz -C 0 -o /Users/alex/Library/Caches/Homebrew/poco-1.7.8.tar.gz.incomplete
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 6030k  100 6030k    0     0   241k      0  0:00:24  0:00:24 --:--:--  348k
Downloaded to: /Users/alex/Library/Caches/Homebrew/poco-1.7.8.tar.gz
SHA256: 4e426b4870a79fb42511f6aeadbb688b4047bdc3c9a21ac31deb364ce49ec69f
==> Verifying poco-1.7.8.tar.gz checksum
Warning: Formula reports different SHA256: 0b620887bb24721c72487a17523fb09d81b459eef73ae815bd6e7d308d466dd0
```

**Comment by dunn** [comment]
https://github.com/pocoproject/poco/issues/1640

**Comment by fxcoudert** [comment]
Fix proposed in #11206
